Tests/Unit/MSFT_xWebAppPool.Tests.ps1

```diff
@@ -0,0 +1,862 @@
+
+[xml]$global:PoolCfg = '<add name="DefaultAppPool" queueLength="1000" autoStart="true" enable32BitAppOnWin64="false" managedRuntimeVersion="v4.0" managedRuntimeLoader="webengine4.dll" enableConfigurationOverride="true" managedPipelineMode="Integrated" CLRConfigFile="" passAnonymousToken="true" startMode="OnDemand"><processModel identityType="ApplicationPoolIdentity" userName="" password="" loadUserProfile="true" setProfileEnvironment="true" logonType="LogonBatch" manualGroupMembership="false" idleTimeout="00:20:00" idleTimeoutAction="Terminate" maxProcesses="1" shutdownTimeLimit="00:01:30" startupTimeLimit="00:01:30" pingingEnabled="true" pingInterval="00:00:30" pingResponseTime="00:01:30" logEventOnProcessModel="IdleTimeout" /><recycling disallowOverlappingRotation="false" disallowRotationOnConfigChange="false" logEventOnRecycle="Time, Memory, PrivateMemory"><periodicRestart memory="0" privateMemory="0" requests="0" time="1.05:00:00"><schedule></schedule></periodicRestart></recycling><failure loadBalancerCapabilities="HttpLevel" orphanWorkerProcess="false" orphanActionExe="" orphanActionParams="" rapidFailProtection="true" rapidFailProtectionInterval="00:05:00" rapidFailProtectionMaxCrashes="5" autoShutdownExe="" autoShutdownParams="" /><cpu limit="0" action="NoAction" resetInterval="00:05:00" smpAffinitized="false" smpProcessorAffinityMask="4294967295" smpProcessorAffinityMask2="4294967295" processorGroup="0" numaNodeAssignment="MostAvailableMemory" numaNodeAffinityMode="Soft" /></add>'
+
+if(-not (Test-Path -Path $moduleRoot))
+{
+    $null = New-Item -Path $moduleRoot -ItemType Directory
+}
+else
+{
+    # Copy the existing folder out to the temp directory to hold until the end of the run
+    # Delete the folder to remove the old files.
+    $tempLocation = Join-Path -Path $env:Temp -ChildPath $DSCModuleName
+    Copy-Item -Path $moduleRoot -Destination $tempLocation -Recurse -Force
+    Remove-Item -Path $moduleRoot -Recurse -Force
+    $null = New-Item -Path $moduleRoot -ItemType Directory
+}
+
+Copy-Item -Path $PSScriptRoot\..\..\* -Destination $moduleRoot -Recurse -Force -Exclude '.git'
+
+if (Get-Module -Name $DSCModuleName -All)
+{
+    Get-Module -Name $DSCModuleName -All | Remove-Module
+}
+
+Import-Module -Name $(Get-Item -Path (Join-Path $moduleRoot -ChildPath "$DSCModuleName.psd1")) -Force
+
+if (($env:PSModulePath).Split(';') -ccontains $pwd.Path)
+{
+    $script:tempPath = $env:PSModulePath
+    $env:PSModulePath = ($env:PSModulePath -split ';' | Where-Object {$_ -ne $pwd.path}) -join ';'
+}
+
+
+$executionPolicy = Get-ExecutionPolicy
+if ($executionPolicy -ne 'Unrestricted')
+{
+    Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Force
+    $rollbackExecution = $true
+}
+
+$ModuleName = "MSFT_xWebAppPool"
+
+Import-Module (Join-Path $moduleRoot -ChildPath "DSCResources\$ModuleName\$ModuleName.psm1")
+
+Describe "MSFT_xWebAppPool"{
+    InModuleScope $ModuleName {
+        Function TestTargetResourceSame
```

Comment:
I'm not sure any of these private test functions are really needed, while reviewing this I keep wondering why they are needed. Could you help me understand the thought process behind this function?

I'll make some notes where you can make some logic changes to reduce the dependancies on these and conform to how we've been writing tests in other places (I'm not saying we're doing it right and you're wrong the great thing about open source is being able to learn from each other!)